Basic/ClientDriven/Assets/Scripts/ServerIngredient.cs

```diff
@@ -7,14 +8,19 @@ public class ServerIngredient : ServerObjectWithIngredientType
 
 public enum IngredientType
 {
-    red,
-    blue,
-    purple,
-    max // should be always last
+    Red,
+    Blue,
+    Purple,
+    MAX // should be always last
 }
 
 public class ServerObjectWithIngredientType : NetworkBehaviour
 {
+    [SerializeField]
+    public NetworkVariable<IngredientType> CurrentIngredientType;
```

Comment:
public fields are automatically serialized, so the SerializeField here is redundant (or even dangerous if it causes the serialization o happen twice). I'd suggest to either use `internal` + `SerializeField`, or to drop the `SerializeField`

Reply:
I'd be surprised if public + SerializeField cause the serialization to happen twice, any link to that?

Reply:
I remember somebody from the engine team pointing that out long time ago to me in a PR, but tbh I don't remember for which Unity version that observation was valid (and if it was by design or a bug of a specific major version)

Reply:
just re-asked [here](https://unity.slack.com/archives/C9SQHJGN6/p1669907293851319) to be sure

Reply:
got an answer: "there's no impact in using both [SerializeField] + public". Sorry for the false alarm!